indico_toolkit/indico_wrapper/modelop.py

```diff
@@ -0,0 +1,200 @@
+        self, model_group_id: int, model_id: int | None = None
+    ) -> dict[str, object]:
+        """
+        Return model options for `model_id` of `model_group_id`
+        or the most recent model if `model_id` is not specified.
+        Args:
+            model_group_id (int): id of model group
+            model_id (int, optional): argument to return a specific model within a model group
+        """
+        all_model_options = self.get_all_model_options(model_group_id)
+
+        if model_id is None:
+            return next(all_model_options)
+
+        for model_options in all_model_options:
+            if model_options["id"] == model_id:
+                return model_options
+        else:
+            raise RuntimeError(
+                f"Model group {model_group_id} does not have a model with ID {model_id}"
+            )
+
+    def get_all_model_options(self, model_group_id: int) -> Iterator[dict[str, object]]:
+        """
+        Return model options for all models of `model_group_id`.
+        Args:
+            model_group_id (int): id of model group
+        Returns:
+            dict: dictionary of all model options, including:
+            "id", "domain, ""high_quality", "interlabeler_resolution",
+            "sampling_strategy", "seed", "test_split", "weight_by_class_frequency",
+            "word_predictor_strength", and "model_training_options"
+        """
+
+        response = self.client.call(
+            GraphQLRequest(
+                """
+                query getMoonbowModelGroupModels($modelGroupId: Int!) {
+                    modelGroup(modelGroupId: $modelGroupId) {
+                        id
+                        models {
+                            id
+                            modelOptions
+                        }
+                    }
+                }
+                """,
+                {"modelGroupId": model_group_id},
+            )
+        )
+
+        for model in response["modelGroup"]["models"]:
+            model_options = json.loads(model["modelOptions"])
+            model_options["id"] = model["id"]
+            yield model_options
+
+    def retrain_model(self, model_group_id: int) -> dict:
+        """
+        Forces model retrain.
+        Args:
+            model_group_id (int): id of model group needing retraining
+        Returns:
+            dict: returns model group id and status of retraining model
+        """
+        return self.client.call(
+            GraphQLRequest(
+                """
+                mutation retrainMoonbowModelGroup($modelGroupId: Int!) {
+                    retrainModelGroup(modelGroupId: $modelGroupId, forceRetrain: true) {
+                        id
+                        status
+                    }
+                }    
+                """,
+                {"modelGroupId": model_group_id},
+            )
+        )
+
+    def update_model_settings(
+        self, model_group_id: int, model_type: str, **kwargs
```

Comment:
I'd probably name this `model_training_options` just to make it a little more descriptive.

```suggestion
        self, model_group_id: int, model_type: str, **model_training_options
```